backend/controllers/file_controller.py

```diff
@@ -0,0 +1,28 @@
+import os
+from werkzeug.utils import secure_filename
+import config
+from flask_restful import Resource
+from flask import request, abort
+
+
+def allowed_file(filename):
+    return ('.' in filename and
+            filename.rsplit('.', 1)[1].lower() in config.ALLOWED_EXTENSIONS)
+
+
+class File(Resource):
+    def post(self):
+        if 'uploaded_data' not in request.files:
+            abort(500)
```

Comment:
Should this be 4xx errors as they are caused by invalid user input? Also I think abort() can take an optional error message, that might be useful in the debug console logs.